catalog/rest/catalog-rest-service/src/main/java/org/codice/ddf/rest/service/CatalogService.java

```diff
@@ -92,6 +97,9 @@ String addDocument(
       InputStream message)
       throws CatalogServiceException;
 
+  Map.Entry<AttachmentInfo, Metacard> parseParts(
```

Comment:
✏️ This is really weird to return a `Map.Entry` in a context that has nothing to do with maps.

Reply:
I chose it because it's in the JDK and implemented by `lang3.Pair`, so it matched the existing code better.

Reply:
still, we are driving the interface based on the implementation and as I said, there is no map context here